### DISCUSSION TITLE
How do I input exact time for transactions?

### DISCUSSION THREAD

**Discussion by Ashesh3** [firefly-iii/firefly-iii#7485] [Firefly III] [answered]
Seems like inputting the exact time for a transaction was a thing, but I am unable to find it anywhere. Was it removed?

![image](https://github.com/firefly-iii/firefly-iii/assets/3626859/eaa97e72-d3c1-49ab-93ea-b9444ec9892a)

Related issue: https://github.com/firefly-iii/firefly-iii/issues/4578

```
Debug information generated at 2023-05-11 00:00:00 UTC for Firefly III version **6.0.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.9 |
| Firefly III API | 2.0.1 |
| Build | 466, 29-04-2023 06:55:07 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |
```

**Comment by JC5** [comment, marked as answer]
This was a reference to the new layout I'm working on where this will be possible.

**Reply by Ashesh3** [reply]
> This was a reference to the new layout I'm working on where this will be possible.

Thanks for your work, the new UI looks really good! I don't believe in ETAs so just keep up the good work :)

**Comment by Ashesh3** [comment]
@JC5 Something interesting I noticed, It appears to me that the backend already supports time, along with date for transactions. 

So currently we have `<input autocomplete="off" name="date[]" type="date" placeholder="Date" title="Date" class="form-control">`
which renders like this:
![image](https://github.com/firefly-iii/firefly-iii/assets/3626859/6e1e8721-92e3-47e5-8139-893c48355569)

If i just change the input type to `datetime-local`, I can now add the time and it's even saved in the backend:
`<input autocomplete="off" name="date[]" type="datetime-local" placeholder="Date" title="Date" class="form-control">`
which renders like this:
![image](https://github.com/firefly-iii/firefly-iii/assets/3626859/3fe0697d-8245-4809-97ab-7a85e3c89294)

Saving it with time, even reflects back in other places:
![image](https://github.com/firefly-iii/firefly-iii/assets/3626859/cd30a5d2-54eb-4662-b1dd-1ee119c073d2)


So do you think just changing the input type could be a temporary solution here?

**Reply by JC5** [reply]
I know the backend supports it, but I want to leave the current layout as it is.

**Reply by edmundo096** [reply]
Is there any way to have this `input type to datetime-local` working by a config/env var/feature flag? (even if it's experimental like Chromium's)
I like the fact it would be using browser's native input control.

_Off topic:_ So far I have been enjoying Firefly albeit the little features that have been changed/removed in the past months because small issues and not reaching a great UX as far I have read. Thank you for your great attention to the project JC5, specially on the tickets attention.

**Reply by Ashesh3** [reply]
@JC5 ^ , It'd be great to have this useful future, especially considering that the backend already has support for it! How's the new UI going? :D
